src/test/java/com/coveo/nashorn_modules/ModuleTest.java

```diff
@@ -548,4 +548,12 @@ public void itIncludesFilenameInException() throws Throwable {
       assertEquals("bad thing in /file1.js at line number 3", e.getMessage().substring(0, 39));
     }
   }
+
+  @Test
+  public void itLoadsIndexJsIfMainDirSpecified() throws Throwable {
```

Comment:
Nitpicking, but there are already a few tests that cover various use cases like this around line 188; it'd be cool to have this one in the same spot. Also, those tests use mocks instead of static resources to setup each use case. Maybe do the same here? It requires almost no code makes the test a bit easier to understand I think.